dhall/src/Dhall/Eval.hs

```diff
@@ -800,6 +806,7 @@ quote !env !t =
                                        NaturalFoldCl{}           -> quote env (t VPrimVar)
                                        ListFoldCl{}              -> quote env (t VPrimVar)
                                        OptionalFoldCl{}          -> quote env (t VPrimVar)
+                                       NaturalSubtractZero       -> App NaturalSubtract (NaturalLit 0)
```

Comment:
Not using `t` here feels rather funny. Should I add a `VNaturalSubtractZero` constructor to `Val` instead?

Reply:
What `t` are you referring to?  This looks correct to me

Reply:
The `t` from `VHLam i t`.